common/src/main/java/com/wynntils/core/features/Configurable.java

```diff
@@ -6,11 +6,22 @@
 
 import com.wynntils.core.config.ConfigHolder;
 import java.util.List;
+import java.util.Optional;
 
 public interface Configurable {
     void updateConfigOption(ConfigHolder configHolder);
 
+    /** Registers the configurable's config options. Called by ConfigManager when loaded */
     void addConfigOptions(List<ConfigHolder> options);
 
+    /** Returns all configurable options registered that should be visible to the user */
```

Comment:
Here we can actually see another benefit of doing interface/abstract base class separation. Now it is clear that to be able to implement Configurable, all these other methods are needed as well. This was not obvious before. Nice!